sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/xml/XmlSuite.scala

```diff
@@ -3515,6 +3524,103 @@ class XmlSuite
   }
 }
 
+class XmlSuiteWithOptimizedXMLParser extends XmlSuite {
+  import testImplicits._
+
+  override protected val memoryEfficientParserEnabled: Boolean = true
```

Comment:
nit: Suggest doing it the other way around since the new optimized parser is going to be default.

```suggestion
class XmlSuiteWithLegacyParser extends XmlSuite {
  import testImplicits._

  override protected val memoryEfficientParserEnabled: Boolean = false
```